player.py

```diff
@@ -125,29 +133,51 @@ def __iter__(self):
 
 
 class CaptainState(State):
-    def __init__(self, player, game):
-        super().__init__(player, game)
-        self.board_str = player.get_board_str(game)
+    def __init__(self, can_act, is_game_stopped, board_str):
+        super().__init__(can_act, is_game_stopped)
+        self.board_str = board_str
+
+    @classmethod
+    def from_player(cls, player, game):
+        state = State.from_player(player, game)
+        return cls(state.can_act, state.is_game_stopped, player.get_board_str(game))
 
 
 class FirstMateState(State):
-    def __init__(self, player, game):
-        super().__init__(player, game)
-        self.powers_charges = player.get_powers_charges()
-        self.hp = player.submarine.hp
+    def __init__(self, can_act, is_game_stopped, powers_charges, hp):
+        super().__init__(can_act, is_game_stopped)
+        self.powers_charges = powers_charges
+        self.hp = hp
+
+    @classmethod
+    def from_player(cls, player, game):
+        state = State.from_player(player, game)
+        return cls(state.can_act, state.is_game_stopped, player.get_powers_charges(), player.submarine.hp)
 
 
 class EngineerState(State):
-    def __init__(self, player, game):
-        super().__init__(player, game)
-        self.tools_state = player.get_tools_state()
+    def __init__(self, can_act, is_game_stopped, tools_state):
+        super().__init__(can_act, is_game_stopped)
+        self.tools_state = tools_state
+
+    @classmethod
+    def from_player(cls, player, game):
+        state = State.from_player(player, game)
+        return cls(state.can_act, state.is_game_stopped, player.get_tools_state())
     
 
 class RadioOperatorState(State):
-    def __init__(self, player, game):
-        super().__init__(player, game)
+    def __init__(self, is_game_stopped, last_enemy_move_direction):
+        super().__init__(True, is_game_stopped)
         del self.__dict__["can_act"]
+        self.last_enemy_move_direction = last_enemy_move_direction
+
+    @classmethod
+    def from_player(cls, player, game):
+        state = State.from_player(player, game)
+        del state.__dict__["can_act"]
```

Comment:
Why use this `del`?
Why not keep `can_act` with constant value of True

Reply:
Because them it will be part of the return value of ``__iter__``  function, and there is no need to send a const True from the server to the client